gems/smithy/lib/smithy/templates/client/types.erb

```diff
@@ -8,6 +8,7 @@ module <%= namespace %>
 <% types.each do |type| -%>
```

Comment:
Not sure if this cleans up more, but instead of types, we can have both structures and unions separately and iterate both lists.